### PR TITLE
apply status_code to HttpResponse

### PR DESCRIPTION
**Pull request by euikook**
Apply status_code of content  to status argument of HttpResponse

### CODE REVIEW

File: django_ajax/response.py [hunk 1]
```diff
@@ -27,5 +27,6 @@ def __init__(self, data, *args, **kwargs):
 
         super(JSONResponse, self).__init__(
             content=serialize_to_json(data, *args, **kwargs),
+            status=data.get('status', 200),
```
Comment (yceruto) [inline django_ajax/response.py]: You can describe the use case where it's useful?

**Comment by yceruto** [comment]
My idea is that all responses can be handled by `@ajax` decoration (including exception). So in theory this one should be `200` always.

**Comment by yceruto** [comment]
Closing as all responses already have status 200.

**Comment by paolobb4** [comment]
Sorry, I normally avoid raising the dead but here's my two cents on this.

HTTP already has status and statusText, why should this info be repeated within the response's content?
Right now from JavaScript's side the response looks something like this:
```yml
XMLHttpRequest {
    ⋯
    status: 200;
    statusText: "OK";
    response: { # Redundant wrapper around actual content
        status: 200;
        satusText: "OK";
        content: { ⋯ }; # Actual json content
    };
    ⋯
}
```
As you can see this produces a redundant layer of content wich only serves to make js's XMLHttpResponse.onerror useless, wich is normaly used to handle non 200 responses (wich in my opinion helps make code more readable and compact).

I propose to remove this redundancy remaining with a simple and clean:
```yml
XMLHttpRequest {
    ⋯
    status: 200;
    satusText: "OK";
    response: { ⋯ }; # json content
    ⋯
}
```


Also, why have it always respond with 200? ajax should be able to handle a non 200 status. I can't debug my deployed application because django-ajax is catching all errors and I'm not getting the ADMIN emails :/  (I'm now thinking of sending the error back from client side just to report it @_@)

Now that I've typed it out I realize that maybe this two questions are really the same.

**Pull request closed without merging**